go.js

```diff
@@ -87,6 +87,11 @@ export async function main(ns) {
     const opponent = ["Netburners", "Slum Snakes", "The Black Hand", "Tetrads", "Daedalus", "Illuminati"];
     const opponent2 = ["Netburners", "Slum Snakes", "The Black Hand", "Tetrads", "Daedalus", "Illuminati", "????????????"];
 
+    if (ns.getResetInfo() == 8) {

```

Comment:
Could you please confirm whether adding this increases the RAM cost of this script?

The solution is fine, but if it increases the footprint I'll want to get the current bitnode using the ram-dodging approach in helpers.js.

Reply:
Agreed, it's a mess in there.

Best way is to find an example from another file (many need a similar check).

In this case, it looks like there's no dedicated helper, we just wrap the call:
```suggestion
    let resetInfo = await getNsDataThroughFile(ns, 'ns.getResetInfo()');
    if (resetInfo.currentNode == 8) {
```

_(Good that we checked, it looks like your PR had a bug - it was comparing 8 to the entire ResetInfo object rather than the '.currentNode' property)_

----

**Advanced**
Since we're only getting that one property, the above should suffice, but for future readers, you'll note that some usages do something like this instead:
```javascript
    let resetInfo = (/**@returns{ResetInfo}*/() => undefined)();
    resetInfo = await getNsDataThroughFile(ns, 'ns.getResetInfo()');
    if (resetInfo.currentNode == 8) {
```
This can be helpful because ram-dodging through a file loses TypeScript information you normally get by using NS api functions directly. This is a trick to restore type hints without actually switching everything from .js to .ts scripts.